test/distributed/checkpoint/test_state_dict.py

```diff
@@ -187,11 +193,39 @@ def init_model_optim():
 
             if compile_model:
                 dist_model = torch.compile(dist_model)
-            dist_optim = torch.optim.Adam(dist_model.parameters(), lr=1e-3)
+            dist_optim = optimizer_class(dist_model.parameters(), lr=1e-3)
```

Comment:
For FSDP2, we may need to pass `foreach=True` to get closer numerics with a non-FSDP2 model. Maybe that might be contributing some numeric difference for AdamW?